File: 347-TopKFrequentElements/347-TopKFrequentElements.cpp
// Last updated: 8/29/2026, 7:02:50 PM
1class Solution {
2public:
3    vector<int> topKFrequent(vector<int>& nums, int k) {
4        map<int,int> mp;
5        for(int i:nums)
6        {
7            mp[i]++;
8        }
9        vector<pair<int,int>> pr;
10        for(auto &i:mp)
11        {
12            pr.push_back({i.first,i.second});
13        }
14        sort(pr.begin(),pr.end(),[](auto &a,auto &b)
15        {
16            if(a.second == b.second)
17            {
18                return a.first<b.first;
19            }
20            return a.second>b.second;
21        });
22        vector<int> ans;
23        for(int i = 0;i<k;i++)
24        {
25            ans.push_back(pr[i].first);
26        }
27        return ans;
28    }
29};

File: 347-TopKFrequentElements/347-TopKFrequentElements.py
# Last updated: 8/29/2026, 7:19:48 PM
1class Solution:
2    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
3        d = {}
4        for i in nums:
5            d[i] = d.get(i,0)+1
6        newitems = sorted(d.items(),key = lambda x: (x[1],x[0]),reverse = True)
7        l = []
8        for i in range(k):
9            l.append(newitems[i][0])
10        return l